src/Cnab.Bundle/Parameter.cs

```diff
@@ -27,42 +50,82 @@ public class ParameterMetadata
         public string Description { get; set; }
     }
 
-    public class StringParameter : Parameter
+    public class StringParameter : Parameter<string>
     {
-        [JsonProperty("defaultValue")]
-        public string DefaultValue { get; set; }
-
-        [JsonProperty("allowedValues")]
-        public List<string> AllowedValues { get; set; }
-
         [JsonProperty("minLength")]
         public int MinimumLength { get; set; }
 
         [JsonProperty("maxLength")]
         public int MaximumLength { get; set; }
-    }
 
-    public class BoolParameter : Parameter
-    {
-        [JsonProperty("defaultValue")]
-        public bool DefaultValue { get; set; }
+        public override bool IsValid(string value)
+        {
+            if (!this.AllowedValues.Contains(value))
+                return false;
 
-        [JsonProperty("allowedValues")]
-        public List<bool> AllowedValues { get; set; }
+            if ((value.Length < this.MinimumLength) || (value.Length > this.MaximumLength))
```

Comment:
What are `this.MinimumLength` and `this.MaximumLength` if not specified in the JSON?  Won't they be 0 since the properties are non-nullable?  Will this then do what you expect?